hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rsql/RSQLUtility.java

```diff
@@ -171,13 +187,20 @@ public Predicate toPredicate(final Root<T> root, final CriteriaQuery<?> query, f
         private final Root<T> root;
         private final CriteriaBuilder cb;
         private final Class<A> enumType;
+        private final VirtualPropertyLookup virtualPropertyLookup;
+        private final StrSubstitutor substitutor;
 
         private final SimpleTypeConverter simpleTypeConverter;
 
-        private JpqQueryRSQLVisitor(final Root<T> root, final CriteriaBuilder cb, final Class<A> enumType) {
+        private JpqQueryRSQLVisitor(final Root<T> root, final CriteriaBuilder cb, final Class<A> enumType,
+                VirtualPropertyLookup virtualPropertyLookup) {
             this.root = root;
             this.cb = cb;
             this.enumType = enumType;
+            this.virtualPropertyLookup = virtualPropertyLookup;
```

Comment:
If you use the `virtualPropertyLookUp` directly because it's extending the `StrLookup` see comment below, then this comment is obsolete.